pyogrio/tests/test_core.py

```diff
@@ -611,3 +613,137 @@ def test_error_handling_warning(capfd, naturalearth_lowres):
         read_info(naturalearth_lowres, INVALID="YES")
 
     assert capfd.readouterr().err == ""
+
+
+def test_vsimem_listtree(naturalearth_lowres):
+    # Prepare test data in /vsimem
+    meta, _, geometry, field_data = read(naturalearth_lowres)
+    meta["spatial_index"] = False
+    meta["geometry_type"] = "MultiPolygon"
+    test_file_path = Path(f"/vsimem/pyogrio_test_{naturalearth_lowres.stem}.gpkg")
```

Comment:
```suggestion
    test_file_path = Path("/vsimem/pyogrio_test_naturalearth_lowres.gpkg")
```

(slight preference to simplify and hardcode filename name)